src/utils.jl

```diff
@@ -120,6 +120,38 @@ function imextendedmin(img::AbstractArray; h::Int=2, conn::Int=2)::BitMatrix
     return Bool.(mask_minima)
 end
 
+"""
+    impose_minima(I::AbstractArray{T}, BW::AbstractArray{Bool}) where {T<:Integer}
+
+Use morphological reconstruction to enforce minima on the input image `I` at the positions where the binary mask `BW` is non-zero.
+
+It supports both integer and grayscale images using different implementations for each.
+"""
+function impose_minima(I::AbstractArray{T}, BW::AbstractArray{Bool}) where {T<:Integer}
```

Comment:
minor detail, but consider adding a comment after `where {T<:Integer} #color image`. 
Then below on line 139, consider `where {T<:AbstractFloat} #gray image`. 
It's clear in the docstring, so maybe not needed.